src/Git.hs

```diff
@@ -143,7 +143,7 @@ data TagResult
   deriving stock (Eq, Show)
 
 data GitOperationFree a
-  = FetchBranch Branch a
+  = FetchBranch Branch Bool a
```

Comment:
This boolean is blind, meaning that the meaning of it is not clear. Could you make a separate data type? e.g. `data FetchWithTags = FetchWithTags | DoNotFetchWithTags`?

Reply:
Right, but I thought that since we never operate directly on `ADT` - it's always lifted via respective function, a naked `Bool` would suffice

Reply:
Fixed in 8a7e805